sale_purchase_requisition/models/sale_order.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    purchase_agent = fields.Many2one(
+        comodel_name="res.users", string="Comercial compra"
+    )
+
+    purchase_requisition_ids = fields.One2many(
+        comodel_name="purchase.requisition",
+        inverse_name="sale_order_id",
+        string="Acuerdo de compra",
+    )
+
+    def action_view_sale_requisition(self):
+        return {
+            "name": _("Purchase Requisition"),
+            "type": "ir.actions.act_window",
+            "res_model": "purchase.requisition",
+            "view_mode": "form",
+            "res_id": self.purchase_requisition_ids.id,
+        }
+
+    def _create_lines_preq(self):
+        self.ensure_one()
+        pur_req_lines = []
+        for line in self.order_line:
+            if line.product_id.purchase_ok:
+                pur_req_lines.append(
+                    [
+                        0,
+                        0,
+                        {
+                            "product_id": line.product_id.id,
+                            "product_qty": line.product_qty,
+                            "product_uom_id": line.product_uom.id,
+                        },
+                    ]
+                )
+        return pur_req_lines
+
+    def action_create_purchase_requisition(self):
+        pur_req_lines = self._create_lines_preq()
+        context_ps = {
+            "default_sale_order_id": self.id,
+            "default_line_ids": pur_req_lines,
+            "default_origin": self.name,
+            "default_user_id": (self.user_id or self.env.user).id,
+            "default_date_end": self.validity_date,
+        }
+        if self.purchase_agent:
+            context_ps["default_user_id"] = self.purchase_agent.id
+        return {
+            "name": _("Purchase Requisition"),
+            "type": "ir.actions.act_window",
+            "res_model": "purchase.requisition",
+            "view_mode": "form",
+            "res_id": self.purchase_requisition_ids.id,
+            "context": context_ps,
+        }
+
+    def write(self, vals):
+        res = super().write(vals)
+        for rec in self:
+            if "order_line" in vals and rec.state in {"draft", "sent"}:
+                for pr in rec.purchase_requisition_ids:
+                    pr.line_ids.unlink()
```

Comment:
This will block the order modifications because unlinking purchase requisition lines when are not in draft, throws an UserError: https://github.com/odoo/odoo/blob/14.0/addons/purchase_requisition/models/purchase_requisition.py#L157